app/controllers/PaperlessPreferencesController.scala

```diff
@@ -16,28 +16,36 @@
 
 package controllers
 
+import com.google.inject.Inject
 import config.ConfigDecorator
 import controllers.auth._
 import controllers.auth.requests.UserRequest
-import com.google.inject.Inject
+import play.api.Mode.Mode
 import play.api.i18n.{Messages, MessagesApi}
 import play.api.mvc.{Action, AnyContent}
-import services.partials.PreferencesFrontendPartialService
+import play.api.{Configuration, Environment}
+import uk.gov.hmrc.play.config.ServicesConfig
 import uk.gov.hmrc.renderer.{ActiveTabMessages, TemplateRenderer}
-import util.LocalPartialRetriever
+import util.{LocalPartialRetriever, Tools}
 
 import scala.concurrent.Future
 
 class PaperlessPreferencesController @Inject()(
   val messagesApi: MessagesApi,
-  val preferencesFrontendPartialService: PreferencesFrontendPartialService,
+  environment: Environment,
+  configuration: Configuration,
+  tools: Tools,
   authJourney: AuthJourney,
   withActiveTabAction: WithActiveTabAction,
   withBreadcrumbAction: WithBreadcrumbAction)(
   implicit partialRetriever: LocalPartialRetriever,
   configDecorator: ConfigDecorator,
   templateRenderer: TemplateRenderer)
-    extends PertaxBaseController {
+    extends PertaxBaseController with ServicesConfig {
+
+  val mode: Mode = environment.mode
+  val runModeConfiguration: Configuration = configuration
+  val preferencesFrontendUrl = baseUrl("preferences-frontend")
```

Comment:
Can this be moved to ConfigDecorator? It would mean you don't have to extend with ServicesConfig

Reply:
done

Reply:
done